File: LeetCode/21.Merge_Two_Sorted_Lists/21.Merge_Two_Sorted_Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy = new ListNode();

        ListNode n = dummy, n1 = list1, n2 = list2;
        while (n1 != null && n2 != null) {
            if (n1.val <= n2.val) {
                n.next = n1;
                n1 = n1.next;
                n = n.next;
            } else {
                n.next = n2;
                n2 = n2.next;
                n = n.next;
            }
        }

        if (n1 != null) {
            n.next = n1;
        }

        if (n2 != null) {
            n.next = n2;
        }

        return dummy.next;
    }
}

File: LeetCode/21.Merge_Two_Sorted_Lists/21.Merge_Two_Sorted_Lists.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func mergeTwoLists(_ list1: ListNode?, _ list2: ListNode?) -> ListNode? {
        let dummy = ListNode(-1)
        var cur: ListNode? = dummy
        var l1: ListNode? = list1, l2: ListNode? = list2

        while let a = l1, let b = l2 {
            if a.val <= b.val {
                cur?.next = a
                l1 = a.next
            } else {
                cur?.next = b
                l2 = b.next
            }
            cur = cur?.next
        }

        cur?.next = l1 ?? l2

        return dummy.next
    }
}
